client/fingerprint/network.go

```diff
@@ -200,14 +199,57 @@ func (f *NetworkFingerprint) findInterface(deviceName string) (*net.Interface, e
 		return nil, err
 	}
 
+	// Since we aren't given an interface to use we have to pick using a
+	// heuristic. To choose between interfaces we avoid down interfaces, those
+	// that are loopback, point to point, or don't have an address. Of the
+	// remaining interfaces we rank them positively for each global unicast
+	// address and slightly negatively for a link local address. This makes
+	// us select interfaces that have more globally routable address over those
+	// that do not.
+	var bestChoice *net.Interface
+	bestScore := -1000
 	for _, intf := range intfs {
-		if f.isDeviceEnabled(&intf) && !f.isDeviceLoopBackOrPointToPoint(&intf) && f.deviceHasIpAddress(&intf) {
-			interfaces = append(interfaces, intf)
+		// We require a non-loopback interface that is enabled with a valid
+		// address.
+		if !f.isDeviceEnabled(&intf) || f.isDeviceLoopBackOrPointToPoint(&intf) || !f.deviceHasIpAddress(&intf) {
+			continue
+		}
+
+		addrs, err := f.interfaceDetector.Addrs(&intf)
+		if err != nil {
+			return nil, err
 		}
-	}
 
-	if len(interfaces) == 0 {
-		return nil, nil
+		seenLinkLocal := false
+		intfScore := 0
+		for _, addr := range addrs {
+			// Find the IP Addr and the CIDR from the Address
+			var ip net.IP
+			switch v := (addr).(type) {
+			case *net.IPNet:
+				ip = v.IP
+			case *net.IPAddr:
+				ip = v.IP
+			}
+
+			if !seenLinkLocal && (ip.IsLinkLocalUnicast() || ip.IsLinkLocalMulticast()) {
+				intfScore -= 1
+				seenLinkLocal = true
+			}
+
+			if ip.IsGlobalUnicast() {
+				intfScore += 2
+			}
+		}
+
+		if intfScore > bestScore {
```

Comment:
If bestscore stays at zero this will pick the first `intf` from the `intfs` list, is that acceptable?